diff -r 1b997a96b779 src/gui/mainFrame.cpp
--- src/gui/mainFrame.cpp	Sat Mar 23 17:50:44 2013 +0000
+++ src/gui/mainFrame.cpp	Sat Mar 23 17:56:36 2013 +0000
@@ -2546,9 +2546,9 @@
 	string s;
 	s=locateDataFile("3Depict-manual.pdf");
 
-	//Also Debian makes us use the lowercase "D", so check there too.
-	if(!s.size())
-		s=locateDataFile("3depict-manual.pdf");
+	//Also Fedora has diff dir
+	if(!wxFileExists(wxStr(s)))
+		s="/usr/share/doc/3Depict-0.0.8/3Depict-0.0.8-manual.pdf";
 
 
 	//If we found it, use the default program associated with that data file
