diff -r 7abb69436c2b src/wx/wxcomponents.cpp
--- src/wx/wxcomponents.cpp	Sat Aug 02 05:39:24 2014 -0400
+++ src/wx/wxcomponents.cpp	Sat Aug 02 05:40:51 2014 -0400
@@ -547,16 +547,17 @@
 	//(Oh look Ma, I'm autoconf!)
 
 	const char *dirs[] = {	".",
-				"/usr/share/fonts/truetype", //Old debian 
+				"/usr/local/share/fonts/truetype", // User fonts
 				"/usr/share/fonts/truetype/freefont", // New debian
 				"/usr/share/fonts/truetype/ttf-dejavu", //New debian
-				"/usr/local/share/fonts/truetype", // User fonts
+				"/usr/share/fonts/truetype", //Old debian 
+				"/usr/share/fonts/dejavu", //Fedora
 				"/usr/X11R6/lib/X11/fonts/truetype",
 				"/usr/X11R6/lib64/X11/fonts/truetype",
-				"/usr/lib/X11/fonts/truetype",// Fedora 32
-				"/usr/lib64/X11/fonts/truetype", //Fedora 64
-				"/usr/local/lib/X11/fonts/truetype", // Fedora 32 new
-				"/usr/local/lib64/X11/fonts/truetype",// Fedora 64 new
+				"/usr/lib/X11/fonts/truetype",
+				"/usr/lib64/X11/fonts/truetype", 
+				"/usr/local/lib/X11/fonts/truetype", 
+				"/usr/local/lib64/X11/fonts/truetype",
 				"",
 				}; //MUST end with "".
 
