Documentation directory change for debian
Index: 3depict/src/3Depict.cpp
===================================================================
--- 3depict.orig/src/3Depict.cpp	2012-11-14 22:44:44.000000000 +0100
+++ 3depict/src/3Depict.cpp	2012-11-14 22:44:47.000000000 +0100
@@ -2651,11 +2651,13 @@
 {
 	//First attempt to locate the local copy of the manual.
 	string s;
-	s=locateDataFile("3Depict-manual.pdf");
+	s=locateDataFile("manual.pdf");
 
 	//Also Debian makes us use the lowercase "D", so check there too.
 	if(!s.size())
-		s=locateDataFile("3depict-manual.pdf");
+	{
+		s=std::string("/usr/share/3depict/manual.pdf");
+	}
 
 
 	//If we found it, use the default program associated with that data file
