Description: Debian packages are lowercase, but upstream uses uppercase
 "D"
Forwarded: not-needed
Author: D Haley <mycae - gmx - com>
--- 3depict-0.0.16.orig/packaging/3Depict.desktop
+++ 3depict-0.0.16/packaging/3Depict.desktop
@@ -2,8 +2,8 @@
 Version=1.0
 Type=Application
 Comment=Valued point cloud visualisation and analysis
-TryExec=3Depict
-Exec=3Depict %F
-Icon=3Depict
-Name=3Depict
+TryExec=3depict
+Exec=3depict %F
+Icon=3depict
+Name=3depict
 
