diff -r 8ddea1426c31 packaging/3Depict.desktop
--- a/packaging/3Depict.desktop	Sat Mar 26 20:09:37 2011 +0000
+++ b/packaging/3Depict.desktop	Sat Mar 26 20:16:19 2011 +0000
@@ -2,8 +2,8 @@
 Version=1.0
 Type=Application
 Comment=Valued point cloud visualisation and analysis
-TryExec=3Depict
-Exec=3Depict %F
-Icon=3Depict
-Name=3Depict
+TryExec=3depict
+Exec=3depict %F
+Icon=3depict
+Name=3depict
 
