Index: 3depict-0.0.1/packaging/3Depict.desktop
===================================================================
--- 3depict-0.0.1.orig/packaging/3Depict.desktop	2010-08-09 23:48:57.000000000 +1000
+++ 3depict-0.0.1/packaging/3Depict.desktop	2010-08-09 23:49:08.000000000 +1000
@@ -2,8 +2,8 @@
 Version=1.0
 Type=Application
 Comment=Valued point cloud visualisation and analysis
-TryExec=3Depict
-Exec=3Depict %F
-Icon=3Depict
-Name=3Depict
+TryExec=3depict
+Exec=3depict %F
+Icon=3depict
+Name=3depict
 
