Index: 3depict-0.0.5/src/assertion.h
===================================================================
--- 3depict-0.0.5.orig/src/assertion.h	2011-03-26 20:27:01.000000000 +0000
+++ 3depict-0.0.5/src/assertion.h	2011-03-26 20:27:03.000000000 +0000
@@ -21,14 +21,15 @@
 		std::cerr << "ASSERTION ERROR!" << std::endl;
 		std::cerr << "Filename: " << filename << std::endl;
 		std::cerr << "Line number: " << lineNumber << std::endl;
-
-		std::cerr << "Do you wish to continue?(y/n)";
+		std::cerr << "well, we are going to continue anyway, but please report this, as it indicates a fault with the program..." << std::endl
+/*		std::cerr << "Do you wish to continue?(y/n)";
 		char y = 'a';
 		while (y != 'n' && y != 'y')
 			std::cin >> y;
 
 		if (y != 'y')
 			exit(1);
+*/
 	}
 
 	inline void dh_warn(const char * const filename, const unsigned int lineNumber,const char *message) 
