Index: 3depict-0.0.6/src/basics.cpp
===================================================================
--- 3depict-0.0.6.orig/src/basics.cpp	2011-05-31 23:13:06.000000000 +0100
+++ 3depict-0.0.6/src/basics.cpp	2011-05-31 23:13:11.000000000 +0100
@@ -48,7 +48,7 @@
 //Program name
 const char *PROGRAM_NAME = "3Depict";
 //Program version
-const char *PROGRAM_VERSION = "0.0.5";
+const char *PROGRAM_VERSION = "0.0.6";
 //Path to font for Default FTGL  font
 const char *FONT_FILE= "FreeSans.ttf";
 
