Upstream assumes different font directories than debian.
Index: 3depict/src/wxcomponents.cpp
===================================================================
--- 3depict.orig/src/wxcomponents.cpp	2012-11-14 22:44:44.000000000 +0100
+++ 3depict/src/wxcomponents.cpp	2012-11-14 22:45:13.000000000 +0100
@@ -681,8 +681,15 @@
 {
 	//This is a list of possible target dirs to search
 	//(Oh look Ma, I'm autoconf!)
+
+	//DEBIAN PATCH: So, it looks like someone moved all the fonts! 
+	// I have no idea why they did that... Lets just use the dirs I have on my
+	// clean debian machine. I have to find a better way to do this, or please
+	// stop moving my fonts!
 	const char *dirs[] = {	".",
 				"/usr/share/fonts/truetype",
+				"/usr/share/fonts/truetype/freefont",
+				"/usr/share/fonts/truetype/ttf-dejavu",
 				"/usr/local/share/fonts/truetype",
 				"/usr/X11R6/lib/X11/fonts/truetype",
 				"/usr/X11R6/lib64/X11/fonts/truetype",
