Debian uses different text domain for the lang files
Index: 3depict-0.0.13/src/3Depict.cpp
===================================================================
--- 3depict-0.0.13.orig/src/3Depict.cpp	2013-03-23 16:07:50.000000000 +0100
+++ 3depict-0.0.13/src/3Depict.cpp	2013-03-23 16:08:54.000000000 +0100
@@ -164,7 +164,7 @@
 		else
 		{
 			//Set the gettext language
-			textdomain( PROGRAM_NAME );
+			textdomain( "3depict");
 			setlocale (LC_ALL, "");
 #ifdef __WXMAC__
 			bindtextdomain( PROGRAM_NAME, paths->GetResourcesDir().mb_str(wxConvUTF8) );
@@ -196,8 +196,8 @@
 					break;
 			}			
 #else
-			bindtextdomain( PROGRAM_NAME, "/usr/share/locale" );
-			bind_textdomain_codeset(PROGRAM_NAME, "utf-8");
+			bindtextdomain( "3depict", "/usr/share/locale" );
+			bind_textdomain_codeset("3depict", "utf-8");
 #endif
 		}
 	}
