Debian uses different text domain for the lang files
Index: 3depict-0.0.8/src/3Depict.cpp
===================================================================
--- 3depict-0.0.8.orig/src/3Depict.cpp	2011-10-23 16:32:18.000000000 +0100
+++ 3depict-0.0.8/src/3Depict.cpp	2011-10-23 16:32:18.000000000 +0100
@@ -347,7 +347,7 @@
 		else
 		{
 			//Set the gettext language
-			textdomain( PROGRAM_NAME );
+			textdomain( "3depict");
 			setlocale (LC_ALL, "");
 #ifdef __WXMAC__
 			bindtextdomain( PROGRAM_NAME, paths->GetResourcesDir().mb_str(wxConvUTF8) );
@@ -371,8 +371,8 @@
 			}
 			
 #else
-			bindtextdomain( PROGRAM_NAME, "/usr/share/locale" );
-			bind_textdomain_codeset(PROGRAM_NAME, "utf-8");
+			bindtextdomain( "3depict", "/usr/share/locale" );
+			bind_textdomain_codeset("3depict", "utf-8");
 #endif
 		}
 	}
