Index: 3depict-0.0.3/src/basics.h
===================================================================
--- 3depict-0.0.3.orig/src/basics.h	2010-09-22 20:26:26.000000000 +0100
+++ 3depict-0.0.3/src/basics.h	2010-09-22 20:26:31.000000000 +0100
@@ -107,7 +107,7 @@

 	//	  an application installer
 	//Mac
 	//	- Look in cwd
-	return std::string(name);
+	return std::string("/usr/share/3depict/") + std::string(name);
 }
 
 template<class T1, class T2>
