Index: 3depict-0.0.7/src/wxcommon.cpp
===================================================================
--- 3depict-0.0.7.orig/src/wxcommon.cpp	2011-07-28 21:57:42.000000000 +0100
+++ 3depict-0.0.7/src/wxcommon.cpp	2011-07-28 21:57:45.000000000 +0100
@@ -66,7 +66,7 @@
 
 #ifdef __linux__
 
-	return std::string("/usr/share/3Depict/") + std::string(name);
+	return std::string("/usr/share/3depict/") + std::string(name);
 #else
 
 	//E.g. Mac
