Index: 3depict-0.0.5/src/wxcommon.cpp
===================================================================
--- a/src/wxcommon.cpp	Sat Mar 26 20:09:37 2011 +0000
+++ b/src/wxcommon.cpp	Sat Mar 26 20:15:58 2011 +0000
@@ -47,7 +47,7 @@
 
 #ifdef __linux__
 
-	return std::string("/usr/share/3Depict/") + std::string(name);
+	return std::string("/usr/share/3depict/") + std::string(name);
 #else
 
 	//E.g. Mac
