From 41d62b5da42806fe814df4c9b36585d37ab01657 Mon Sep 17 00:00:00 2001
From: D Haley <mycae@gmx.com>
Date: Mon, 3 Oct 2016 22:32:31 +0100
Subject: * Fix compilation error where C++11 feature used, when compiling in
 C++98


diff --git a/src/backend/filters/lukasAnalysis.cpp b/src/backend/filters/lukasAnalysis.cpp
index ece8fff..01513ba 100755
--- a/src/backend/filters/lukasAnalysis.cpp
+++ b/src/backend/filters/lukasAnalysis.cpp
@@ -46,7 +46,8 @@ enum
 
 LukasAnalysisFilter::LukasAnalysisFilter() : 
 	rsdIncoming(0)
-{
+{ 
+	vdbCache= openvdb::FloatGrid::create(0.0);
 
 	rgba=ColourRGBAf(0.5,0.5,0.5,1.0f);
 	iso_level=0.07;
diff --git a/src/backend/filters/lukasAnalysis.h b/src/backend/filters/lukasAnalysis.h
index c560d9d..06c3226 100755
--- a/src/backend/filters/lukasAnalysis.h
+++ b/src/backend/filters/lukasAnalysis.h
@@ -56,7 +56,7 @@ class LukasAnalysisFilter : public Filter
 		
 		//Cache to use for vdbgrid info
 		// console warning: non-static data member initializers only available with -std=c++11 or -std=gnu++11
-		openvdb::FloatGrid::Ptr vdbCache = openvdb::FloatGrid::create(0.0);
+		openvdb::FloatGrid::Ptr vdbCache;
 
 	public:
 		//!Constructor
